crates/language-server/src/db.rs

```diff
@@ -71,6 +84,28 @@ impl LanguageServerDatabase {
         map_file_to_mod(self, file)
     }
 
+    pub fn find_enclosing_item(&mut self, top_mod: TopLevelMod, cursor: Cursor) -> Option<ItemKind> {
```

Comment:
If I understood correctly, it's unnecessary to use lazy spans here, but I couldn't remember how to find the regular item spans so I just tried lazy spans to see if I could get it to work that way

Reply:
I think it'd be better to extract lazy spans and then evaluate them to `Span` since 
1.  each item owns span information as `HirOrigin<T>` to abstract lowering and macro expansion effects, which is a low-level representation and difficult to handle
2. it's efficient enough.

Reply:
By the way, I guess it'd be better to track the function that collects all items span in the `top_mod` so that we don't need to rerun it.